src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -1,70 +1,67 @@
-            add(new ProductDto(2L, "płaszcz", "Pellentesque(...)", new BigDecimal(150), 1L));
-            add(new ProductDto(3L, "buty", "Pellentesque(...)", new BigDecimal(100), 4L));
-            add(new ProductDto(4L, "rękawiczki", "Pellentesque(...)", new BigDecimal(50), 2L));
-        }
-    };
+    @Autowired
+    ProductService productService;
+    @Autowired
+    ProductMapper productMapper;
 
     @GetMapping
     public ResponseEntity<List<ProductDto>> getProducts() {
-        return ResponseEntity.ok(testProductList);
+        List<Product> productList = productService.getAllProducts();
+        return ResponseEntity.ok(productMapper.mapToProductDtoList(productList));
     }
 
     @GetMapping("{productId}")
     public ResponseEntity<ProductDto> getProduct(@PathVariable Long productId) {
-        for (ProductDto product : testProductList) {
-            if (product.getProductId().equals(productId)) {
-                return ResponseEntity.ok(product);
-            }
+        if (productService.getProduct(productId) != null) {
+            return ResponseEntity.ok(productMapper.mapToProductDto(productService.getProduct(productId)));
+        } else {
+            return ResponseEntity.notFound().build();
         }
-        return ResponseEntity.notFound().build();
     }
 
     @DeleteMapping(value = "{productId}")
     public ResponseEntity<Void> deleteProduct(@PathVariable Long productId) {
-        for (ProductDto product : testProductList) {
-            if (product.getProductId().equals(productId)) {
-                testProductList.remove(product);
-                return ResponseEntity.ok().build();
-            }
+        if (productService.getProduct(productId) != null) {
+            productService.deleteById(productId);
+            return ResponseEntity.ok().build();
+        } else {
+            return ResponseEntity.notFound().build();
         }
-        return ResponseEntity.notFound().build();
     }
 
     @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<Void> createProduct(@RequestBody ProductDto productDto) {
-        testProductList.add(productDto);
-
+        Product product = productMapper.mapToProduct(productDto);
+        productService.saveProduct(product);
         return ResponseEntity.ok().build();
     }
 
     @PutMapping(path = "{productId}", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<ProductDto> updateProduct(@PathVariable Long productId, @RequestBody ProductDto productNew) {
-        for (ProductDto product : testProductList) {
-            if (product.getProductId().equals(productId)) {
-                testProductList.remove(product);
-                testProductList.add(productNew);
-                return ResponseEntity.ok(productNew);
-            }
+    public ResponseEntity<ProductDto> updateProduct(@PathVariable Long productId, @RequestBody ProductDto newProductDto) {
```

Comment:
I don't know why yet, but I checked on postman - with each product update, the id is changed by 1. Because the id is a reference point for other classes - it needs to be fixed.

Reply:
I found similar bug in my User classes and I suspect that the same is causing the bug in yours.
Try to delete  productService.deleteById(productId) in line 60 and check if it work.